README.md

```diff
@@ -55,22 +74,53 @@ To change the path of the end-point, use the `-web.collectd-push-path` command
 line option. To disable this functionality altogether, use
 `-web.collectd-push-path=""`.
 
-## Using Docker
 
-You can deploy this exporter using the [prom/collectd-exporter][hub] Docker image.
-You will need to map the collectd port from the container to the host, remembering
-that this is a UDP port.
+## Build
 
-For example:
+To be able to build from source code, you need to have a working Go environment with [verion 1.7 or greater](https://golang.org/doc/install) installed.
+
+Follow the steps here to build using `make`:
+
+    $ mkdir -p $GOPATH/src/github.com/prometheus
+    $ cd $GOPATH/src/github.com/prometheus
+    $ git clone git@github.com:cch0/collectd_exporter.git
+    $ cd collectd_exporter
+    $ make all
+
+The Makefile provides several targets:
+
+  * *build*: build the binary
+  * *format*: format the source code
+  * *docker*: build a docker container for the current `HEAD`
+  * all: execute `format`, `build`, `docker` goals.
+
+
+### Build Artifact
+
+* A executable binary `collectd_exporter` should exist after the build
+* A Docker image `tmobile/collectd-exporter` should exist in your local docker registry.
+
+### Note
+* Depends on the platform you are using to build the binary, the artifact is only capable of running on the platform you build it with.
+
+* Use the `go env` command to find out which `architecture` and `os` you are building with. The following configuration is for buidling to run on Linux platform.
+
+	`GOARCH="amd64"`
+	
+	`GOOS="linux"`
+
+
+
+## Run
+
+To run the executable binary from the command line
+
+```$./collectd_exporter```
+
+To run the Docker container
+
+```docker run -d  -p 9103:9103 -p 25826:25826/udp tmobile/collectd-exporter -collectd.listen-address=":25826" -metadata.refresh.period.min="1"```
 
-```bash
```

Comment:
Depends on how this binary going to be executed. Running it as binary is possible but not as good as running it as docker container assuming all their ec2 instances can run docker. I will add the docker section if that is the direction we are going with in the future.